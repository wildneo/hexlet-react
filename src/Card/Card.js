import React from 'react';

export default class Card extends React.Component {
  render() {
    const { title, text } = this.props;
    return (
      <div className="card">
        <div className="card-body">
          <h4 className="card-title">{title}</h4>
          <p className="card-text">{text}</p>
        </div>
      </div>
    );
  }
}
