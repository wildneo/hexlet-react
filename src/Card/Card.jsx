import React from 'react';

const Card = ({ children }) => (<div className="card">{children}</div>);
const Body = ({ children }) => (<div className="card-body">{children}</div>);
const Title = ({ children }) => (<h4 className="card-title">{ children }</h4>);
const Text = ({ children }) => (<p className="card-text">{ children }</p>);

Card.Body = Body;
Card.Title = Title;
Card.Text = Text;

export default Card;
