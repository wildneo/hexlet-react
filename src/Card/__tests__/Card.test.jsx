import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';

test('Card', () => {
  const dom = <Card />;
  const component = renderer.create(dom);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Card 2', () => {
  const dom = (
    <Card>
      <Card.Title>Title</Card.Title>
    </Card>
  );
  const component = renderer.create(dom);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Card 3', () => {
  const dom = (
    <Card>
      <Card.Body>
        <Card.Title>Title</Card.Title>
        <Card.Text>Text</Card.Text>
      </Card.Body>
    </Card>
  );
  const component = renderer.create(dom);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Card 4', () => {
  const dom = (
    <Card>
      <Card.Body>
        <Card.Title>Title 3</Card.Title>
        <Card.Text>Text 3</Card.Text>
      </Card.Body>
    </Card>
  );
  const component = renderer.create(dom);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
