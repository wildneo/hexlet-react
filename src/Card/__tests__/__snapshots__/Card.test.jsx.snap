// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<div
  className="card"
/>
`;

exports[`Card 2 1`] = `
<div
  className="card"
>
  <h4
    className="card-title"
  >
    Title
  </h4>
</div>
`;

exports[`Card 3 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      Title
    </h4>
    <p
      className="card-text"
    >
      Text
    </p>
  </div>
</div>
`;

exports[`Card 4 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      Title 3
    </h4>
    <p
      className="card-text"
    >
      Text 3
    </p>
  </div>
</div>
`;
